src/async-input.js

```diff
@@ -39,6 +39,27 @@ export class AsyncInput {
   getValue() {}
 }
 
+/**
+ * Attributes
+ *
+ * Components implementing the AsyncInput,
+ * are expected to assert these attributes
+ * at the appropriate time.
+ *
+ * @constant {!Object<string, string>}
+ */
+export const AsyncInputAttributes = {
+  /**
+   * data-name
+   *
+   * Required attribute that must be asserted by every async-input
+   * Element. This is used by AMP form to add the key
+   * for the form submission request
+  */
+  'NAME': 'data-name',
```

Comment:
design nit: data attributes are typically used when creating new attributes for existing DOM elements, like `input` or `p`. This prevents new standards and specs from the W3C from being added to existing DOM elements like `input` and making your markup behave differently over time.

Since you're making a custom element `amp-recaptcha-input` that the W3C will never add new attributes to, the `data-*` prefix doesn't add the utility that it adds for existing web standard elements.

Reply:
Ahh so since I am passing things through a 3p frame, it seems that all data you want to be passed through the 3p frame, we have been prepending with `data-*`. See https://www.ampproject.org/docs/reference/components/amp-gist . and we already have `data-sitekey` and `data-action`

Originally I thought it would be better to be consistent. But now that I realize that layout is also not `data-`, I can agree with this 😄 Thanks!